mlp/preprocess_data.py

```diff
@@ -0,0 +1,157 @@
+    if not os.path.exists(folder):
+        os.makedirs(folder)
+    with open(filename, 'wb') as csvfile:
+        writer = csv.writer(csvfile)
+        writer.writerows(data)
+
+
+def write_dataset_to_csv(dataset, output_directory):
+    """Write dataset to multiple csv files"""
+    csv = dataset.to_csv_data()
+    for output, csv_data in csv:
+        final_path = os.path.join(output_directory, output + ".csv")
+        print "Writing", len(csv_data), "samples: ", final_path
+        write_to_csv(final_path, csv_data)
+
+
+def load_examples(self, path, label_mapper):
+    """
+    Load examples contained in the path into an example collection. Examples need to be stored in CSVs.
+
+    The label_mapper allows to map a loaded label to a different label, e.g. to combine multiple labels into one.
+
+    Arguments:
+
+    :param path: can be directory or zipfile. If zipfile, it will be extracted to a temporary path with prefix ``/tmp/muvr-training-``
+    :param label_mapper: the mapper
+    :return:
+    """
+    self.label_id_mapping = {}
+    root_directory = ""
+    if os.path.isdir(path):
+        root_directory = path
+    else:
+        # Zip file - extract to temp root_directory first
+        root_directory = tempfile.mkdtemp(prefix="/tmp/muvr-training-")
+        zipfile.ZipFile(path, 'r').extractall(root_directory)
+
+    #csv_files = filter(lambda f: f.endswith("csv"), os.listdir(root_directory))
+    csv_files = self.read_all_csv(root_directory)
+
+    xs = []
+    ys = []
+    for f in csv_files:
+        for label, x in self.load_example(f, label_mapper):
+            if label not in self.label_id_mapping:
+                self.label_id_mapping[label] = len(self.label_id_mapping)
+            xs.append(x)
+            ys.append(self.label_id_mapping[label])
+
+    return ExampleColl(xs, ys)
+
+def load_data(path, label_mapper):
+    """ Load all csv files with converting label
+        Return list of object ExampleColl, each has the same label
+    """
+    root_directory = ""
+    if os.path.isdir(path):
+        root_directory = path
+    else:
+        # Zip file - extract to temp root_directory first
+        root_directory = tempfile.mkdtemp(prefix="/tmp/muvr-training-")
+        zipfile.ZipFile(path, 'r').extractall(root_directory)
+
+    csv_files = CSVAccelerationDataset.read_all_csv(root_directory)
+
+    label_mapping = {}
+    for f in csv_files:
+        for label, x in CSVAccelerationDataset.load_example(f, label_mapper):
+            data = label_mapping.get(label, [])
+            data.append(x)
+            label_mapping[label] = data
+
+    result = []
+    for label in label_mapping:
+        xs = label_mapping[label]
+        ys = [label] * len(xs)
+        result.append(ExampleColl(xs, ys))
+    return result
+
+
+def main(dataset_directory, output_directory, ratio, is_slacking):
+    """Main entry point."""
+
+    # 1/ Load the dataset
+    dataset = load_data(dataset_directory, map_label)
```

Comment:
Alright, let me rephrase: Why do you want to clone the dataset with your labels instead of doing the mapping on the fly?